src/main/java/picard/illumina/SampleInputParameters.java

```diff
@@ -0,0 +1,100 @@
+package picard.illumina;
+
+import htsjdk.samtools.util.StringUtil;
+import picard.illumina.parser.ReadDescriptor;
+import picard.illumina.parser.ReadStructure;
+import picard.illumina.parser.ReadType;
+import picard.util.IlluminaUtil;
+import picard.util.TabbedTextFileWithHeaderParser;
+
+import java.io.File;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class SampleInputParameters {
+    /**
+     * Column header for the first barcode sequence (preferred).
+     */
+
+    public static final Set<String> BARCODE_PREFIXES = new HashSet<>(
+            Arrays.asList(SampleInputParameters.BARCODE_SEQUENCE_COLUMN, SampleInputParameters.BARCODE_COLUMN)
+    );
+
+    public static final String BARCODE_COLUMN = "barcode";
+    public static final String BARCODE_SEQUENCE_COLUMN = "barcode_sequence";
+
+    /**
+     * Column header for the barcode name.
+     */
+    public static final String BARCODE_NAME_COLUMN = "barcode_name";
+    /**
+     * Column header for the library name.
+     */
+    public static final String LIBRARY_NAME_COLUMN = "library_name";
+
+
+    protected static void parseInputFile(final File inputFile,
```

Comment:
1. Could use javadoc.
2. Does picard use any library that has a decent `Either` such that we could return an Either with the payload or the list of error messages?  That'd be nicer than taking mutable parameters.  If not could use `Tuple` from htsjdk to return two things
3. It's a bit weird to me that this returns/makes `BarcodeMetric` instances - are those always the thing thats needed?  Maybe we could define a little `BarcodeMetadata` class, return that, and have it have a `toBarcodeMetric()` method to build a metric object when it's needed?

Reply:
For 3. I'm pretty sure we only populate the metrics map and that's all we do with the information coming back from the method. So if we did create a `BarcodeMetadata` object we then turn around and call `toBarcodeMetric()` on it and then nothing else, but I'll double check.